linux_kernel_modules/rtc_sync/rtc_sync.c

```diff
@@ -0,0 +1,176 @@
+/*
+ * rtc_sync.c: module that syncs sytem time with rtc time 
+ * and vice versa with 10 seconds interval
+ *
+ * Copyright (C) 2018 Sierra Wireless.
+ *
+ * This file is licensed under the terms of the GNU General Public License
+ * version 2.  This program is licensed "as is" without any warranty of any
+ * kind, whether express or implied.
+ */
+
+/* IMPORTANT: the RTC only stores whole seconds. It is arbitrary
+ * whether it stores the most close value or the value with partial
+ * seconds truncated. However, it is important that we use it to store
+ * the truncated value. This is because otherwise it is necessary,
+ * in an rtc sync function, to read both xtime.tv_sec and
+ * xtime.tv_nsec. On some processors (i.e. ARM), an atomic read
+ * of >32bits is not possible. So storing the most close value would
+ * slow down the sync API. So here we have the truncated value and
+ * the best guess is to add 0.5s.
+ */
+
+#include <linux/module.h>
+#include <linux/rtc.h>
+#include <linux/timer.h>
+#include <linux/time.h>
+#include <linux/sched.h>
+#include <linux/kthread.h>
+
+
+static struct task_struct *time_check_thread;
+
+
+//this function syncs system time with RTC when startup - if there is valid time store in RTC
+static int rtc_hctosys(void)
+{
+	int err;
+	struct rtc_time tm;
+	struct rtc_device *rtc = rtc_class_open("rtc1");
+	if (rtc == NULL) {
+		printk("%s: unable to open rtc device (rtc1) for rtc_hctosys\n",
+			__FILE__);
+		return -ENODEV;
+	}
+
+	err = rtc_read_time(rtc, &tm);
+	if (err == 0) {
+		err = rtc_valid_tm(&tm);
+		if (err == 0) {
+			struct timespec tv;
+
+			tv.tv_nsec = NSEC_PER_SEC >> 1;
+
+			rtc_tm_to_time(&tm, &tv.tv_sec);
+
+			do_settimeofday(&tv);
+
+			dev_info(rtc->dev.parent,
+				"setting system clock to "
+				"%d-%02d-%02d %02d:%02d:%02d UTC (%u)\n",
+				tm.tm_year + 1900, tm.tm_mon + 1, tm.tm_mday,
+				tm.tm_hour, tm.tm_min, tm.tm_sec,
+				(unsigned int) tv.tv_sec);
+		} else { 
+			dev_err(rtc->dev.parent,
+				"hctosys: invalid date/time\n");
+		}
+	} else {
+		dev_err(rtc->dev.parent,
+			"hctosys: unable to read the hardware clock\n");
+	}
+
+	rtc_class_close(rtc);
+
+	return 0;
+}
+
+//systohc sets RTC with system time
+static int systohc(struct timespec now)
+{
+	
+        struct rtc_device *rtc;
```

Comment:
This line is indented with spaces.  There are other lines that are indented with spaces as well.  Kernel code should be indented with tabs only.